### PR TITLE
Move JComponentRecord to namespace

### PR DESCRIPTION
**Pull request by laoneo**
Moves JComponentRecord to namespace and changes some old doc blocks to the new set up. Additionally it fixes the ComponentHelper class which has some code which isn't adapted to namespaces, probably because of an invalid merge.

### CODE REVIEW

File: libraries/src/Joomla/Cms/Component/ComponentRecord.php [hunk 1]
```diff
@@ -1,12 +1,13 @@
 <?php
 /**
- * @package     Joomla.Libraries
- * @subpackage  Component
+ * Joomla! Content Management System
  *
- * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
- * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ * @copyright  Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE
```
Comment (brianteeman) [inline libraries/src/Joomla/Cms/Component/ComponentRecord.php]: please lets not repeat the mistake of elsewhere the file is LICENSE.txt

**Comment by laoneo** [comment]
Changed all files then in libraries/src/Joomla/Cms from license to license.txt

**Comment by brianteeman** [comment]
thank you @laoneo

**Comment by wilsonge** [comment]
Sorry I fixed all the ComponentHelper stuff earlier when i did a staging -> 3.8 -> 4.0 merge. You're going to need to fix conflicts

**Comment by laoneo** [comment]
Done.

**Pull request merged**